src/lib/eina/eina_thread_posix.c

```diff
@@ -0,0 +1,312 @@
+   sigaddset(&newset, SIGALRM);
+   sigaddset(&newset, SIGCHLD);
+   sigaddset(&newset, SIGUSR1);
+   sigaddset(&newset, SIGUSR2);
+   sigaddset(&newset, SIGHUP);
+   sigaddset(&newset, SIGQUIT);
+   sigaddset(&newset, SIGINT);
+   sigaddset(&newset, SIGTERM);
+# ifdef SIGPWR
+   sigaddset(&newset, SIGPWR);
+# endif
+   pthread_sigmask(SIG_BLOCK, &newset, &oldset);
+#endif
+   err = pthread_create((pthread_t *)t, &attr, func, data);
+#ifndef _WIN32
+   pthread_sigmask(SIG_SETMASK, &oldset, NULL);
+#endif
+   pthread_attr_destroy(&attr);
+
+   if (err == 0) return EINA_TRUE;
+
+   return EINA_FALSE;
+}
+
+static inline Eina_Bool
+_eina_thread_equal(Eina_Thread t1, Eina_Thread t2)
+{
+   return pthread_equal((pthread_t)t1, (pthread_t)t2);
+}
+
+static inline Eina_Thread
+_eina_thread_self(void)
+{
+   return (Eina_Thread)pthread_self();
+}
+
+
+typedef struct _Eina_Thread_Call Eina_Thread_Call;
+struct _Eina_Thread_Call
+{
+   Eina_Thread_Cb func;
+   const void *data;
+
+   Eina_Thread_Priority prio;
+   int affinity;
+};
+
+static void *
+_eina_internal_call(void *context)
+{
+   Eina_Thread_Call *c = context;
+   void *r;
+   pthread_t self;
+
+   // Default this thread to not cancellable as per Eina documentation
+   eina_thread_cancellable_set(EINA_FALSE, NULL);
+
+   EINA_THREAD_CLEANUP_PUSH(free, c);
+
+   self = pthread_self();
+
+   if (c->prio == EINA_THREAD_IDLE)
+     {
+        struct sched_param params;
+        int min;
+#ifdef SCHED_IDLE
+        int pol = SCHED_IDLE;
+#else
+        int pol;
+        pthread_getschedparam(self, &pol, &params);
+#endif
+        min = sched_get_priority_min(pol);
+        params.sched_priority = min;
+        pthread_setschedparam(self, pol, &params);
+     }
+   else if (c->prio == EINA_THREAD_BACKGROUND)
+     {
+        struct sched_param params;
+        int min, max;
+#ifdef SCHED_BATCH
+        int pol = SCHED_BATCH;
+#else
+        int pol;
+        pthread_getschedparam(self, &pol, &params);
+#endif
+        min = sched_get_priority_min(pol);
+        max = sched_get_priority_max(pol);
+        params.sched_priority = (max - min) / 2;
+        pthread_setschedparam(self, pol, &params);
+     }
+// do nothing for normal
+//   else if (c->prio == EINA_THREAD_NORMAL)
+//     {
+//     }
+   else if (c->prio == EINA_THREAD_URGENT)
+     {
+        struct sched_param params;
+        int max, pol;
+
+        pthread_getschedparam(self, &pol, &params);
+        max = sched_get_priority_max(pol);
+        params.sched_priority += 5;
+        if (params.sched_priority > max) params.sched_priority = max;
+        pthread_setschedparam(self, pol, &params);
+     }
+
+   _eina_debug_thread_add(&self);
+   EINA_THREAD_CLEANUP_PUSH(_eina_debug_thread_del, &self);
+   r = c->func((void*) c->data, eina_thread_self());
+   EINA_THREAD_CLEANUP_POP(EINA_TRUE);
+   EINA_THREAD_CLEANUP_POP(EINA_TRUE);
+
+   return r;
+}
+
+Eina_Thread
+eina_thread_self(void)
+{
+   return _eina_thread_self();
+}
+
+Eina_Bool
```

Comment:
Why having an internal call definition and an external call on the same file? Why not only having `eina_thread_equal` that does what `_eina_thread_equal` does?

Reply:
The same happens with join, equal and self

Reply:
This is from master. I didn't touch the code.

Reply:
wtf, okay then